solutions/src/main/java/ru/sberbank/school/task10/MyScalableThreadPool.java

```diff
@@ -0,0 +1,189 @@
+package ru.sberbank.school.task10;
+
+import java.util.*;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Future;
+import java.util.concurrent.FutureTask;
+import java.util.stream.IntStream;
+
+/**
+ * Created by Mart
+ * 04.06.2019
+ **/
+public class MyScalableThreadPool implements ThreadPool {
+    private volatile int activeThreads;
+    private final Object activeThreadsLock = new Object();
+    private final int sizeMin;
+    private final int sizeMax;
+    private final Queue<Integer> freeIndexForNewWorker = new LinkedList<>();
+    private final Map<Integer, ThreadWorker> poolMap = new HashMap<>();
+    private final Queue<FutureTask> tasks = new LinkedList<>();
+
+    public MyScalableThreadPool(int sizeMin, int sizeMax) {
+        if (sizeMin <= 0 || sizeMax <= 0 || sizeMax < sizeMin) {
+            throw new IllegalArgumentException("Incorrect input size");
+        }
+
+        this.sizeMin = sizeMin;
+        this.sizeMax = sizeMax;
+        this.activeThreads = 0;
+        fillPoolMap(sizeMin);
+    }
+
+    public MyScalableThreadPool(int fixedSize) {
+        if (fixedSize <= 0) {
+            throw new IllegalArgumentException("Incorrect input size");
+        }
+        this.sizeMin = fixedSize;
+        this.sizeMax = fixedSize;
+        this.activeThreads = 0;
+        fillPoolMap(fixedSize);
+    }
+
+    private void fillPoolMap(int size) {
+        IntStream.range(0, size)
+                .forEach(i -> poolMap.put(i, new ThreadWorker("ThreadWorker " + i, i)));
+
+        IntStream.range(size, sizeMax)
+                .forEach(freeIndexForNewWorker::add);
+    }
+
+    @Override
+    public void start() {
+        poolMap.forEach((key, worker) -> {
+            worker.start();
+
+            synchronized (activeThreadsLock) {
+                activeThreads++;
+            }
+        });
+    }
+
+    @Override
+    public void stopNow() {
+        synchronized (poolMap) {
+            poolMap.forEach((key, worker) -> worker.interrupt());
+            poolMap.clear();
+            freeIndexForNewWorker.clear();
+        }
+        synchronized (tasks) {
+            tasks.clear();
+        }
+        activeThreads = 0;     //  before tasks.clear() sometimes "activeThreads = 1"
+                               // in last reading... after "= 0" writing    NB!!! repeat hb rule !!!
+    }
+
+    @Override
+    public void execute (Runnable runnable){
```

Comment:
WhitespaceAround: &#39;{&#39; is not preceded with whitespace.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a873404469>*